updateFeatureFlag/main.go

```diff
@@ -27,6 +27,8 @@ func init() {
 func updateFeatureFlag(flagId string, updateFeatureFlagRequest utils.UpdateFeatureFlagRequest) (events.APIGatewayProxyResponse, error) {
 	db := database.CreateDynamoDB()
```

Comment:
1. Use a single dynamodb client instance. Instead of creating a new dynamodb client instance for each function call, you can create a single instance and reuse it. This will reduce the overhead of creating new connections.
2. Handle errors more gracefully. If there is an error updating the feature flag, you should log the error and return a more informative response to the caller. For example, you could return a http.StatusNotFound response if the feature flag with the given flagId does not exist.
3. Use a custom error type. Instead of returning a generic error value, you could define a custom error type for your function. This will make it easier to handle errors in your code and to provide more informative responses to the caller.

Reply:
For reusing the same instance, I'm thinking to check if the dynamodb instance is not nil, then return the same or else create a new one, **any suggestions**?

Reply:
Yes Tejas we are passing the custom error message to the user, for instance in this case the user will see message like this https://github.com/Real-Dev-Squad/feature-flag-backend/blob/eb767c424d600ac310fa3d5064784449854ab8d2/getFeatureFlagById/main.go#L23

Reply:
For the Internal errors like `some error related to database` we are logging the exact error and for the user showing an `Internal Server error`, so that we don't share the user much internal information